site-manage/datemanager/impl/src/main/java/org/sakaiproject/datemanager/impl/DateManagerServiceImpl.java

```diff
@@ -1398,7 +1399,8 @@ public DateManagerValidation validateForums(String siteId, JSONArray forums) thr
 
                                 String entityType = (String)jsonForum.get(DateManagerConstants.JSON_EXTRAINFO_PARAM_NAME);
                                 DateManagerUpdate update;
-                                if(resourceLoader.getString("itemtype.forum").equals(entityType)) {
+								String normalizedType = (entityType == null) ? "unknown" : entityType.toLowerCase();						
+								if (normalizedType.startsWith(TYPE_FORUMS)) {
```

Comment:
so is the translated string important or not important? does this work in non-USA locales?

Reply:
Thank you - You are both correct. I will work up an internationalized solution and re-submit the pull request

Reply:
this code seems a bit crazy with saving internationalized strings into JSON. that's obviously very wrong, but i fear that modifying it will be a big undertaking. and you seem to be looking to fix a clear, small bug.

Reply:
I won't have time to test this refactor right away but was considering something like this:

String entityType = (String) jsonForum.get(DateManagerConstants.JSON_EXTRAINFO_PARAM_NAME);
DateManagerUpdate update;

String forumLabel = resourceLoader.getString("itemtype.forum");           // e.g., "Forums"
String draftLabel = resourceLoader.getString("itemtype.draft");           // e.g., "Draft"
String draftForumLabel = forumLabel + " " + draftLabel;                   // e.g., "Forums Draft"

if (forumLabel.equalsIgnoreCase(entityType) || draftForumLabel.equalsIgnoreCase(entityType)) {
    BaseForum forum = forumManager.getForumById(true, forumId);
    if (forum == null) {

Reply:
@ottenhoff We are moving forward locally with the code in the pull request as we need a fix this for our 23.x production instance. Given that the presence of internationalized in the extrainfo fields seems wrong would you be able to open a separate jira to account for that and go forward with this pull request fix?   I don't know when I will be able to get back to this to add internationalization to the this fix as we are heavily involved into seeing if we can move to 25.x before Turnitin support drops for 23.x in Fall 2026.

Reply:
But your PR as-is wont work for anyone except English locales, right? That's a non-starter for this project.

Reply:
just get rid of your hard-coded english, and i'm sure this PR is fine

`if(entityType.startsWith(resourceLoader.getString("itemtype.forum"))) {`